spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -66,6 +66,23 @@ def team_id
 
     # Set a new team ID which will be used from now on
     def team_id=(team_id)
+      # First, we verify the team actually exists, because otherwise iTC would return the
+      # following confusing error message
+      #
+      #     invalid content provider id
+      #
+      available_teams = teams.collect do |team|
+        team.fetch("contentProvider", {}).fetch("contentProviderId", nil)
```

Comment:
I would suggest that we stop using `fetch` because I think that it has surprising behavior. It looks like you can give it a default value, but actually that doesn't prevent the value returned from being `nil` if the hash entry contains the value `nil` (which happens a lot in Apple's responses)

I would suggest we use `(team['contentProvider'] || {})` instead.